src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -962,7 +961,7 @@ public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
                         var (type, matchedOverload) = GetFunctionSymbolType(function, syntax.OpenParen, syntax.CloseParen, syntax.Arguments, errors, diagnostics);
                         if (matchedOverload is not null)
                         {
-                            matchedFunctionOverloads.Add(syntax, matchedOverload);
+                            matchedFunctionOverloads.TryAdd(syntax, matchedOverload);
```

Comment:
Are there any issues with `type` & `matchedOverload` not being generated by the same call into GetFunctionSymbolType? Feels like it might be safer for `GetFunctionSymbolType` to handle the interaction with this dictionary.

Reply:
I haven't seen any issues, but it can't really be atomic with two separate concurrent dictionaries being involved unless we add locks or do concurrency differently. I will move the logic inside because all usages of the method were adding the 2nd return value in the tuple to the dictionary.